blocks/editable/index.js

```diff
@@ -507,15 +507,27 @@ export default class Editable extends Component {
 	 * @param {KeydownEvent} event The keydow event as triggered by tinyMCE.
 	 */
 	onKeyDown( event ) {
+		const dom = this.editor.dom;
+		const rootNode = this.editor.getBody();
+
 		if (
 			this.props.onMerge && (
 				( event.keyCode === BACKSPACE && this.isStartOfEditor() ) ||
 				( event.keyCode === DELETE && this.isEndOfEditor() )
 			)
 		) {
-			const forward = event.keyCode === DELETE;
 			this.fireChange();
-			this.props.onMerge( forward );
+
+			const isEmpty = dom.isEmpty( rootNode );
+			if ( this.props.onReplace && isEmpty ) {
+				// Remove this block if the editor is empty
+				this.props.onReplace( [] );
```

Comment:
One odd thing about the logic flow here: If I render an `<Editable />` with `onReplace`, but not `onMerge`, I won't have this emptiness replace behavior, since this is only reached if `onMerge` is assigned, even though it's not called.